File: xskeleton-lib/src/main/java/com/mcgrady/xskeleton/http/interceptor/StringConverter.java
package com.mcgrady.xskeleton.http.interceptor;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;
import com.google.gson.JsonPrimitive;
import com.google.gson.JsonSerializationContext;
import com.google.gson.JsonSerializer;

import java.lang.reflect.Type;

/**
 * Created by mcgrady on 2019-08-30.
 */
public class StringConverter implements JsonSerializer<String>, JsonDeserializer<String> {
    @Override
    public JsonElement serialize(String src, Type typeOfSrc, JsonSerializationContext context) {
        if (src == null) {
            return new JsonPrimitive("");
        } else {
            return new JsonPrimitive(src);
        }
    }

    @Override
    public String deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
            throws JsonParseException {
        return json.getAsJsonPrimitive().getAsString();
    }
}

File: xskeleton-lib/src/main/java/com/mcgrady/xskeleton/http/interceptor/StringConverter.kt
package com.mcgrady.xskeleton.http.interceptor

import com.google.gson.*
import java.lang.reflect.Type

/**
 * Created by mcgrady on 2019-08-30.
 */
class StringConverter : JsonSerializer<String?>, JsonDeserializer<String> {
    override fun serialize(src: String?, typeOfSrc: Type, context: JsonSerializationContext): JsonElement {
        return src?.let { JsonPrimitive(it) } ?: JsonPrimitive("")
    }

    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): String {
        return json.asJsonPrimitive.asString
    }
}